packages/firestore/src/local/local_store.ts

```diff
@@ -971,10 +971,16 @@ export class LocalStore {
   }
 
   // PORTING NOTE: Multi-tab only.
-  getQueryDataForTarget(targetId: TargetId): Promise<QueryData | null> {
-    return this.persistence.runTransaction('Get query data', false, txn => {
-      return this.queryCache.getQueryDataForTarget(txn, targetId);
-    });
+  getQueryForTarget(targetId: TargetId): Promise<Query | null> {
+    if (this.targetIds[targetId]) {
```

Comment:
I wonder if we should rename targetIds to allocatedQueriesByTargetId or similar.  As-is it's not entirely obvious if/when queries will be in targetIds.

Reply:
I had this crazy though that we could just get rid of state... but then I didn't keep pursuing. I added a TODO to rename it to `queryDataByTarget`, which would  match the names in SyncEngine. I didn't want to do this as part of this PR as  I fear that this will create major merge conflicts at some point.